File: lib/src/features/member_staff/models/staff_rating.dart
class StaffRating {
  final int id;
  final int memberId;
  final int staffId;
  final String staffType;
  final int rating;
  final String? feedback;
  final DateTime createdAt;
  final DateTime updatedAt;

  StaffRating({
    required this.id,
    required this.memberId,
    required this.staffId,
    required this.staffType,
    required this.rating,
    this.feedback,
    required this.createdAt,
    required this.updatedAt,
  });

  factory StaffRating.fromJson(Map<String, dynamic> json) {
    return StaffRating(
      id: json['id'],
      memberId: json['member_id'],
      staffId: json['staff_id'],
      staffType: json['staff_type'],
      rating: json['rating'],
      feedback: json['feedback'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'member_id': memberId,
      'staff_id': staffId,
      'staff_type': staffType,
      'rating': rating,
      'feedback': feedback,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}

File: backend-api/app/Models/StaffRating.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class StaffRating extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'member_id',
        'staff_id',
        'staff_type',
        'rating',
        'feedback',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array
     */
    protected $casts = [
        'rating' => 'integer',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Get the member who submitted the rating.
     */
    public function member()
    {
        return $this->belongsTo(Member::class);
    }

    /**
     * Get the staff being rated.
     * This is a polymorphic relationship based on staff_type.
     */
    public function staff()
    {
        if ($this->staff_type === 'society') {
            return $this->belongsTo(SocietyStaff::class, 'staff_id');
        } else {
            return $this->belongsTo(Staff::class, 'staff_id');
        }
    }

    /**
     * Scope a query to only include ratings for a specific staff.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @param  int  $staffId
     * @param  string  $staffType
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function scopeForStaff($query, $staffId, $staffType)
    {
        return $query->where('staff_id', $staffId)
                     ->where('staff_type', $staffType);
    }

    /**
     * Scope a query to only include ratings by a specific member.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @param  int  $memberId
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function scopeByMember($query, $memberId)
    {
        return $query->where('member_id', $memberId);
    }

    /**
     * Scope a query to only include ratings from the last month.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function scopeLastMonth($query)
    {
        return $query->where('created_at', '>=', now()->subMonth());
    }
}
